### ISSUE TITLE
lat lon converters float accuracy

### ISSUE THREAD

**Issue by AllNamesWereTakenIsNotAvailable** [M0r13n/pyais#107]
for the example sentence:
!AIVDM,1,1,,A,100u3g@0291Q1>BW6uDUwDk00LE@,0*74

decode and then encode will results in a change to lat lon

the problem line is:
```
            elif d_type == float:
                val = int(val)
                bits = int_to_bin(val, width, signed=signed)
```

Suggested resolution:
```
            elif d_type == float:
                val = int(round(val))
                bits = int_to_bin(val, width, signed=signed)
```

Kind Regards

**Comment by M0r13n** [comment]
Hey @AllNamesWereTakenIsNotAvailable,

thank you bringing this up. It is also very helpful that you included an example message. So it was easy for me to reproduce your issue.

There was indeed a rounding error. I fixed it in the relevant [conversion function](https://github.com/M0r13n/pyais/pull/108/files#diff-a02771f2d710b2fe42598cb1265860f0c137f2036a1a4dac3eea0efd0ae6eaa0L744).

I also included your example as an unit test to prevent future regressions. So I am closing this for now. Feel free to reopen if needed.

Have a great day.

Kind regards,
leon

**Comment by M0r13n** [comment]
The fix is publicly available with version v2.5.1
